### PR TITLE
Tutorial for private base image update

### PR DESCRIPTION
**Pull request by jaysterp**

### CODE REVIEW

File: docs/Tasks/triggers/private-base-image-update.md [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+
+Only allow the task to pull images from the base image registry.
+
+```
+az role assignment create --assignee $principalID --scope $baseregID --role acrpull
+```
+
+## Add target registry credentials to the task
+
+Add a custom registry credential to the task so that it can authenticate to the base image registry.
```
Comment (northtyphoon) [inline docs/Tasks/triggers/private-base-image-update.md]: Can you point out the custom registry credential is using the task system identity?

Comment (jaysterp) [reply docs/Tasks/triggers/private-base-image-update.md]: Sure I will clarify

File: docs/Tasks/triggers/private-base-image-update.md [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+  --git-access-token $GIT_PAT \
+  --arg REGISTRY_NAME=baseRegistry.azurecr.io \
+  --assign-identity
+```
+
+## Give the identity pull permissions to the base registry
+
+You must grand the managed identity pull permission to the base image registry.
+```
+principalID=$(az acr task show --name hometask --registry homeRegistry --query identity.principalId --output tsv)
```
Comment (SteveLasker) [inline docs/Tasks/triggers/private-base-image-update.md]: Line wrapping for clarity.
Would also suggest nesting these commands, similar to: https://github.com/SteveLasker/Presentations/blob/master/demo-scripts/buffering-building-patching/README.md#automate-build--deploy
It's not like someone would type these, and are copy/pasting anyway. So, might as well link them together in a single copy/paste

File: docs/Tasks/triggers/private-base-image-update.md [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+`ENV NODE_VERSION 9.11.2a`
+
+Now build and push the updated base image to the base registry.
+```
+az acr build --registry baseRegistry --image baseimages/node:9-alpine --file Dockerfile-base .
+```
+
+## List the updated build
+```
+az acr task list-runs --registry homeRegistry --name hometask -o table
```
Comment (SteveLasker) [inline docs/Tasks/triggers/private-base-image-update.md]: Would suggest running this with watch
```sh
watch -n1 az acr task list-runs --registry homeRegistry --name hometask -o table
```

File: docs/Tasks/triggers/private-base-image-update.md [hunk 4]
```diff
@@ -0,0 +1,84 @@
...
+
+```
+az acr task create \
+  --registry homeRegistry \
+  --name hometask \
+  --image helloworld:v2 \
+  --context https://github.com/$GIT_USER/acr-build-helloworld-node.git \
+  --file Dockerfile-app \
+  --git-access-token $GIT_PAT \
+  --arg REGISTRY_NAME=baseRegistry.azurecr.io \
```
Comment (SteveLasker) [inline docs/Tasks/triggers/private-base-image-update.md]: ACR_NAME=${BASE_REGISTRY} \

Comment (jaysterp) [reply docs/Tasks/triggers/private-base-image-update.md]: `REGISTRY_NAME` is an argument in Dockerfile-app. I will address the other comments about env vars and formatting.

**Pull request merged**